pegasus/sites.v3/code.org/public/js/extended_us_map.js

```diff
@@ -789,7 +789,24 @@ function setupMapDrawing($, document, window, Raphael, undefined) {
 
 var webServiceLocation = "/promote/state/";
 
+// Just adding this handler to assist with debugging on production.
+document.addEventListener("DOMContentLoaded", function() {
+  console.log("DOMContentLoaded");
+});
+
 $(document).ready(function () {
+  initExtendedMap();
+});
+
+function initExtendedMap() {
+  if (Raphael) {
+    renderMap();
+  } else {
+    setTimeout(initExtendedMap, 100);
+  }
+};
```

Comment:
I appreciate the persistence, but I'm a bit wary of this approach. the frequent checks could make a slow-loading system even slower, and if the request for Raphael is blocked or fails for other reasons, then we could keep checking forever.

while you could potentially work around those issues, what do you think about taking this as a sign to move this code into apps? It looks like Raphael is available as an npm package: https://github.com/DmitryBaranovskiy/raphael , and npm / yarn take care of making sure that the dependency is fully loaded before you try to rely on it, which seems not to be true when executing our copy of the raphael.js file.

perhaps you can:
1. create an entry point in apps
2. have that page import raphael, and then execute the contents of extended_us_maps.js 
3. use `webpack_asset_path` to load that entry point into this page

this example makes me think that our build system will not complain too much about having this kind of js code within apps: https://github.com/code-dot-org/code-dot-org/blob/17aef2b891b3106a0ae688fb2087b004f6f906cf/apps/src/schoolInfoManager.js#L52

I'd be happy to pair on this tomorrow if that would be helpful.

Reply:
if you are itching to get this wrapped up as-is, I would suggest the following modifications:
1. only poll every 1 second
2. only poll for 5 seconds
3. use `if (typeof Rafael !== 'undefined')`, since what you have looks like it will error if it is not defined:
![Screen Shot 2022-10-20 at 3 49 14 PM](https://user-images.githubusercontent.com/8001765/197077361-2483ff81-1c76-497e-a4f0-5d82a6110bd5.png)

Reply:
4. consider adding a comment saying to first try moving your code to apps/src before copying this pattern 😅

Reply:
Thanks for this thoughtful review, and agreed that moving this code to `/apps` would be the right direction.  

I am also itching to at least get a fix out, since it's been broken for over a week now, so I took your suggestions to at least get this simple patch working, though I gave it up to 10 seconds of retries in case of a particularly slow connection.

Reply:
sounds good, thanks for considering!